### PR TITLE
Add path to duplicate query objects

### PR DESCRIPTION
**Pull request by elpete**
Added logic to duplicate `Query` objects.

If `deep` is true, it also duplicates the Query data.

### CODE REVIEW

File: src/main/java/ortus/boxlang/runtime/types/Query.java [hunk 1]
```diff
@@ -779,4 +780,21 @@ public IStruct getMetaData() {
 		return meta;
 	}
 
+	public Query duplicate() {
```
Comment (lmajano) [inline src/main/java/ortus/boxlang/runtime/types/Query.java:783]: Please add all java docs please

File: src/main/java/ortus/boxlang/runtime/types/Query.java [hunk 2]
```diff
@@ -779,4 +780,21 @@ public IStruct getMetaData() {
...
+		return duplicate( false );
+	}
+
+	public Query duplicate( boolean deep ) {
```
Comment (lmajano) [inline src/main/java/ortus/boxlang/runtime/types/Query.java:787]: javadocs

File: src/main/java/ortus/boxlang/runtime/types/Query.java [hunk 3]
```diff
@@ -779,4 +780,21 @@ public IStruct getMetaData() {
...
+		Query q = new Query();
+		for ( var entry : this.getColumns().entrySet() ) {
```
Comment (lmajano) [inline src/main/java/ortus/boxlang/runtime/types/Query.java:789]: Can you check if the size of the query is > 50 rows and if so, use a parallel stream

File: src/main/java/ortus/boxlang/runtime/util/DuplicationUtil.java [hunk 4]
```diff
@@ -136,4 +137,8 @@ public static Array duplicateArray( Array target, Boolean deep ) {
 		);
 	}
 
+	private static Object duplicateQuery( Query target, Boolean deep ) {
```
Comment (lmajano) [inline src/main/java/ortus/boxlang/runtime/util/DuplicationUtil.java:140]: java docs

File: src/main/java/ortus/boxlang/runtime/types/Query.java (hunk 3 above)
Comment (elpete) [reply src/main/java/ortus/boxlang/runtime/types/Query.java:789]: So this is just adding columns.  So you want that if the query is more than 50 columns?

**Pull request merged**